packages/embed/src/form/form.ts

```diff
@@ -1,33 +1,55 @@
-import { transactionCreated$, formSubmit$ } from '../subjects'
+import FormNapper from 'form-napper'
+import { SubjectManager } from '../subjects'
 
 export type FormNapperInstance = {
   hijack: (fn: CallableFunction) => void
   inject: (name: string, value: string) => void
   submit: () => void
+  detach: () => void
 }
 
+const instances = new Map<string, FormNapperInstance>()
+let instanceCount = 0
+
 export const createFormController = (
-  form: FormNapperInstance,
-  onComplete?: CallableFunction
+  form: HTMLFormElement,
+  onComplete: CallableFunction,
+  subject: SubjectManager
 ) => {
-  form.hijack(() => {
-    formSubmit$.next()
+  // Check if currently managed by form napper
+  if (parseInt(form.dataset.formNapperId) > 0) {
```

Comment:
FormNapper attaches it's own event handlers, we need to check the form and see if we have previously attached an event, if so remove it before attaching another.

Reply:
Would this conflict with the merchants form if they too were using form-napper?

Reply:
Its a data attribute we are adding - we could make it `gr4vyFormId` or something.

Reply:
Ah I see - we could make this more specific but not related to this PR.